libs/components/molecules/src/lib/source-list/source-list.component.ts

```diff
@@ -44,12 +68,34 @@ export class SourceListComponent {
    */
   showTable = true;
 
+  selection = new SelectionModel<SourceListItem>(true, []);
+
+  dataSource = new MatTableDataSource<SourceListItem>();
+
+  displayedColumns: string[] = ['select', 'authors', 'year', 'title', 'doi', 'link'];
+
   /** Emits when the contact button is clicked */
   @Output() readonly collaborateClick = new EventEmitter<void>();
 
+  @Output() readonly selectionChanged = new EventEmitter<SourceListItem[]>();
+
+  @ViewChild(MatSort) sort: MatSort;
+
   /** Google analytics tracking service */
   private readonly ga = inject(GoogleAnalyticsService);
 
+  ngOnChanges(changes: SimpleChanges) {
+    if ('sources' in changes) {
+      this.selection.clear();
+      this.dataSource = new MatTableDataSource<SourceListItem>(this.sources);
```

Comment:
Don't recreate the `dataSource` each time. Instead set `this.dataSource.data = this.sources`